### PR TITLE
Qhc 723 temporal solution to bypass dataloss error

### PR DESCRIPTION
**Pull request by jordivallsq**
Confirmation from HW that this fix solves temporarily the issues with the data loss error.

This changes are temporal as QM fix their code, they are also integrated with the close qprogram fix when an error is thrown

### CODE REVIEW

File: src/qililab/platform/platform.py [hunk 1]
```diff
@@ -837,38 +844,56 @@ def _execute_qprogram_with_quantum_machines(  # pylint: disable=too-many-locals,
         threshold_rotations: dict[str, float | None] = {},
         thresholds: dict[str, float | None] = {},
         calibration: Calibration | None = None,
+        dataloss_tries: int = 3,
```
Comment (DZPM) [inline src/qililab/platform/platform.py:847]: why a different dataloss_tries (3) than before (2)?

File: src/qililab/platform/platform.py [hunk 2]
```diff
@@ -837,38 +844,56 @@ def _execute_qprogram_with_quantum_machines(  # pylint: disable=too-many-locals,
...
+
+            except StreamProcessingDataLossError as dataloss:
+                time_interval = datetime.datetime.now() - start_time
+                start_time = datetime.datetime.now()
+                warnings.warn(
+                    f"Warning: {dataloss} raised, retrying experiment ({iteration+1}/{dataloss_tries} available tries) after {time_interval.seconds} s"
                 )
-                measurement_result.set_classification_threshold(thresholds.get(measurement.bus, None))
-                results.append_result(bus=measurement.bus, result=measurement_result)
+                print(traceback.format_exc())
```
Comment (DZPM) [inline src/qililab/platform/platform.py:886]: Why `print()` ?

As a good practice, I prefer a warning, as warnings logs can be easily disabled.

File: src/qililab/platform/platform.py [hunk 3]
```diff
@@ -837,38 +844,56 @@ def _execute_qprogram_with_quantum_machines(  # pylint: disable=too-many-locals,
...
+                if iteration + 1 != dataloss_tries:
+                    time.sleep(1 * dataloss_tries)
```
Comment (DZPM) [inline src/qililab/platform/platform.py:888]: The way it's implemented, this `sleep(dataloss_tries)` timeout will be part of the `time_interval`.

Not wrong, just take in mind that the real `time_interval` would be a bit shorter.

**Review by DZPM — APPROVED** [review]
Please add at least a test case to validate the `except StreamProcessingDataLossError` block. A mock that rises an error should be enough.

File: src/qililab/platform/platform.py (hunk 1 above)
Comment (jordivallsq) [reply src/qililab/platform/platform.py:847]: This is thee number of loops suggested by HW as too many may be too long on queue (they also use 5 though)

File: src/qililab/platform/platform.py (hunk 2 above)
Comment (jordivallsq) [reply src/qililab/platform/platform.py:886]: great point, fixed!

File: src/qililab/platform/platform.py (hunk 3 above)
Comment (jordivallsq) [reply src/qililab/platform/platform.py:887]: Fixed, I moved the time_start after the wait

**Review by DZPM — APPROVED** [review]
LGTM

**Comment by DZPM** [comment]
Jordi asked me to force merge

**Pull request merged**